code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -23,21 +23,15 @@
 	..()
 	if(stat & NOPOWER)
 		return
-	var/state_change = FALSE
 	if(on)
 		if(beaker)
 			if(!beaker.reagents.total_volume)
 				on = FALSE
-				SSnanoui.update_uis(src)
 				return
 			beaker.reagents.temperature_reagents(desired_temp)
 			beaker.reagents.temperature_reagents(desired_temp)
 			if(abs(beaker.reagents.chem_temp - desired_temp) <= 3)
 				on = FALSE
-			state_change = TRUE
-
-	if(state_change)
-		SSnanoui.update_uis(src)
```

Comment:
If anyone wonders why this is removed. When you call `tgui_act` and return `TRUE` from it, it will automatically push a UI update out without having to wait for the SS to fire or do anything else. This is where most of the responsiveness comes from.

(This is a note, not a requested change. Hell, I approve of this)